### PR TITLE
use tf2 ignition if playing that game

### PR DESCRIPTION
**Pull request by Naleksuh**

### CODE REVIEW

**Comment by peace-maker** [comment]
Why? What's the difference? If you really want this to be merged, you should consider giving a description.

**Comment by Naleksuh** [comment]
> Why? What's the difference? If you really want this to be merged, you should consider giving a description.

it uses the tf2 native when running tf2, which works a lot better than igniteentity imo. it doesn't change anything for other games.

**Comment by psychonic** [comment]
Outside of the functionality changes, you need to check both engine and gamedir (or just appid, if we expose that) if you want to detect a specific game. Just detecting the engine has the possibility for breakage down the road.

**Comment by Naleksuh** [comment]
i'm not sure this code will be possible to use, because now it appears that the tf2.ext is required to load for funcommands to work.

anyone knows a way around this?

**Comment by SlidyBat** [comment]
> i'm not sure this code will be possible to use, because now it appears that the tf2.ext is required to load for funcommands to work, breaking funcommands on any game besides tf2
> 
> anyone knows a way around this?

undef `REQUIRE_EXTENSIONS`

File: plugins/funcommands.sp [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #include <sdktools>
 #undef REQUIRE_PLUGIN
 #include <adminmenu>
-
+#include <tf2>
```
Comment (KyleSanderson) [inline plugins/funcommands.sp]: jannik called this out a couple months ago - can you make these static engine checks to not impact performance substantially by checking upon every invocation?

Comment (Naleksuh) [reply plugins/funcommands.sp]: huh? no, tf2 is included to use TF2_IgnitePlayer

**Comment by KyleSanderson** [comment]
@Naleksuh any update on this one?

**Comment by Naleksuh** [comment]
Okay, so I didn't get an email for the "any update" until this was closed. Can it be reopened?

**Pull request closed without merging**